src/sentry/snuba/metrics/extraction.py

```diff
@@ -1259,7 +1327,7 @@ def _extend_parsed_query(self, parsed_query_result: QueryParsingResult) -> Query
 
         extended_conditions = conditions
         if new_conditions:
-            if self.use_updated_env_logic:
+            if self.spec_version.matches_flags_set({"use_updated_env_logic"}):
```

Comment:
I still think that it's simpler to read to have something like:
```suggestion
            if self.spec_version.has("use_updated_env_logic"):
```
and use Python's boolean operators to combine multiple conditions but I am fine with both.